pyaedt/modules/SolveSetup.py

```diff
@@ -1973,21 +1974,61 @@ def _get_primitives_points_per_net(self):
         net_primitives = edb.modeler.primitives_by_net
         primitive_dict = {}
+        layers_elevation = {
+            lay.name: lay.lower_elevation + lay.thickness / 2
+            for lay in list(self.p_app.modeler.edb.stackup.signal_layers.values())
+        }
         for net, primitives in net_primitives.items():
             primitive_dict[net] = []
-            n = 0
-            while len(primitive_dict[net]) < len(net_primitives[net]):
-                if n > 1000:  # adding 1000 as maximum value to prevent infinite loop
-                    return
-                n += 20
-                primitive_dict[net] = []
-                for prim in primitives:
-                    layer = edb.stackup.signal_layers[prim.layer_name]
-                    z = layer.lower_elevation + layer.thickness / 2
-                    pt = self._get_point_inside_primitive(prim, n)
-                    if pt:
-                        pt.append(z)
-                        primitive_dict[net].append(pt)
+            self.p_app.logger.info("Processing net {}...".format(net))
+            for prim in primitives:
+
+                if prim.layer_name not in layers_elevation:
+                    continue
+                z = layers_elevation[prim.layer_name]
+                if "EdbPath" in str(prim):
+                    points = list(prim.center_line.Points)
+                    pt = [points[0].X.ToDouble(), points[0].Y.ToDouble()]
+                    pt.append(z)
+                    next_p = int(len(points) / 4)
+                    pt = [points[next_p].X.ToDouble(), points[next_p].Y.ToDouble()]
+                    pt.append(z)
+                    primitive_dict[net].append(pt)
+
+                elif "EdbPolygon" in str(prim):
+                    pdata_orig = prim.polygon_data.edb_api
+                    pdata = self.p_app.modeler.edb._edb.Geometry.PolygonData.CreateFromArcs(
+                        pdata_orig.GetArcData(), True
+                    )
+
+                    pdata.Scale(0.99, pdata.GetBoundingCircleCenter())
+                    points = [[], []]
+                    for point in list(pdata.Points):
+                        points[0].append(point.X.ToDouble())
+                        points[1].append(point.Y.ToDouble())
+                    # points = prim.points()
+                    pt = [points[0][0], points[1][0]]
+                    pt.append(z)
+                    primitive_dict[net].append(pt)
+                    next_p = int(len(points[0]) / 4)
+                    pt = [points[0][next_p], points[1][next_p]]
+                    pt.append(z)
+                    primitive_dict[net].append(pt)
+                    next_p = int(len(points[0]) / 2)
+                    pt = [points[0][next_p], points[1][next_p]]
+                    pt.append(z)
+                    primitive_dict[net].append(pt)
+
+                else:
+                    n = 0
+                    while n < 1000:
+                        n += 10
```

Comment:
A week ago we had to switch back to `n += 20`. Do your changes handle the reason for that switch back ?

Reply:
@SMoraisAnsys This is an algorithm issue. @maxcapodi78 figured out how to make it more robust